### PR TITLE
fix: clarify recommended catch all modeling

### PR DESCRIPTION
**Pull request by nikku**
As agreed upon this is to omit the `errorRef` and `escalationRef` for catch-all events.

## Description

Raised through [user feedback](https://forum.camunda.io/t/feedback-modeler-documentation-confusing-for-generic-error-escalation/45829/3?u=nikku) and as [discussed](https://camunda.slack.com/archives/CSQ2E3BT4/p1692183082690069) we want to clearly document one way of catch-all: Omitting `escalationRef` and `errorRef`. This is what is already implemented in our validation tools, and supported by the engine. Just documentation (fixed in this PR) is slightly confusing, to date.

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [x] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [x] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/components/modeler/bpmn/error-events/error-events.md [hunk 1]
```diff
@@ -15,15 +15,14 @@ For example, if an invalid credit card is used in the process below, the process
...
+An error must define an `errorCode`. The value of this `errorCode` is used to determine which catch event can catch the thrown error.
 
 For throwing error events, it is possible to define the `errorCode` as an `expression`. When the event is reached,
 the expression is evaluated. An error with the result of this expression is thrown. If no expression is used the
 statically defined `errorCode` is used.
 
-For error catch events, the `errorCode` can be a static value or it can be left empty. An expression can't be used. A
-catch event with an empty `errorCode` will catch **all** thrown errors.
+For catching error events `errorCode` must be a static value.
+Alternatively a catching error event may omit the error reference all together. In this case it catches **all** thrown errors.
```
Comment (korthout) [inline docs/components/modeler/bpmn/error-events/error-events.md:25]: ❓ @nikku Why change the naming from error catch event to catching error event?

🔧 I suggest keeping `error catch event` for consistency, AFAIK we typically use the BPMN element type name

File: docs/components/modeler/bpmn/error-events/error-events.md [hunk 2]
```diff
@@ -52,8 +51,8 @@ instance.
 
 It is not possible to define multiple error catch events with the same `errorCode` in a single scope. It is also not
 permitted to have multiple error catch events without an `errorCode` in a single scope. The deployment gets rejected in
-these cases. However, it is possible to define both an error catch event **with** an `errorCode` and one **without** an
-`errorCode` in the same scope. When this happens, the error catch event that matches the `errorCode` is prioritized.
+these cases. However, it is possible to define both an error catch event referencing an error with a particular `errorCode` and an error catch-all event within the same scope. When this happens, the error catch event
+that matches the `errorCode` is prioritized.
```
Comment (korthout) [inline docs/components/modeler/bpmn/error-events/error-events.md:55]: 👍 I like the changes, but I think there's another sentence that needs to be adjusted

>It is also not permitted to have multiple error catch events without an `errorCode` in a single scope.

🔧 Is that sentence still relevant? I believe it should say something like:

```
It is also not permitted to have multiple error catch-all events in a single scope.
```

File: docs/components/modeler/bpmn/escalation-events/escalation-events.md [hunk 3]
```diff
@@ -20,14 +20,13 @@ The example above shows the execution of an escalation event:
...
-An escalation must define an `escalationCode`. The value of this `escalationCode` is used to determine which catch event
-can catch the thrown escalation.
+An escalation must define an `escalationCode`. The value of this `escalationCode` is used to determine which catch event can catch the thrown escalation.
 
 For throwing escalation events, it is possible to define the `escalationCode` as an `expression`. When the event is reached, the expression is evaluated.
 An escalation with the result of this expression is thrown. If no expression is used the statically defined `escalationCode` is used.
 
-For catching escalation events it is not possible to use an `expression`. The `escalationCode` must always be a static value.
-Alternatively, the `escalationCode` can be left empty. A catch event with an empty `escalationCode` will catch **all** thrown escalations.
+For catching escalation events `escalationCode` must be a static value.
```
Comment (korthout) [inline docs/components/modeler/bpmn/escalation-events/escalation-events.md:28]: > must be a static value.

💭 We should probably link such statements to https://docs.camunda.io/docs/next/components/concepts/expressions/#expressions-vs-static-values

**Review by korthout — APPROVED** [review]
Thanks @nikku 🙇 

👍 I like the changes 

🔧 and only have some minor suggestions, please have a look before merging

LGTM 👍

File: docs/components/modeler/bpmn/error-events/error-events.md (hunk 1 above)
Comment (nikku) [inline docs/components/modeler/bpmn/error-events/error-events.md:25]:
```suggestion
Alternatively an error catch event may omit the error reference all together. In this case it catches **all** thrown errors.
```

File: docs/components/modeler/bpmn/escalation-events/escalation-events.md (hunk 3 above)
Comment (nikku) [reply docs/components/modeler/bpmn/escalation-events/escalation-events.md:28]: Great idea!

File: docs/components/modeler/bpmn/error-events/error-events.md [hunk 4]
```diff
@@ -15,15 +15,14 @@ For example, if an invalid credit card is used in the process below, the process
...
-error to the error catch events.
+An error must define an `errorCode`. The value of this `errorCode` is used to determine which catch event can catch the thrown error.
 
 For throwing error events, it is possible to define the `errorCode` as an `expression`. When the event is reached,
 the expression is evaluated. An error with the result of this expression is thrown. If no expression is used the
 statically defined `errorCode` is used.
 
-For error catch events, the `errorCode` can be a static value or it can be left empty. An expression can't be used. A
-catch event with an empty `errorCode` will catch **all** thrown errors.
+For catching error events `errorCode` must be a static value.
```
Comment (nikku) [inline docs/components/modeler/bpmn/error-events/error-events.md:24]: One thing I found confusing is that we mix the event (referencing the error) and the actual error. 

```suggestion
Errors that are meant to be caught must have a static `errorCode`.
```

On the other hand I understand that our users may not distinguish between error (can be caught or thrown) and the event (BPMN symbol) that references it.

File: docs/components/modeler/bpmn/escalation-events/escalation-events.md (hunk 3 above)
Comment (nikku) [reply docs/components/modeler/bpmn/escalation-events/escalation-events.md]: I've tried to incorporate this into https://github.com/camunda/camunda-platform-docs/pull/2471/commits/0cea47e66891bcc1389575cefd12018a2472c087.

File: docs/components/modeler/bpmn/error-events/error-events.md (hunk 2 above)
Comment (nikku) [reply docs/components/modeler/bpmn/error-events/error-events.md]: I've tried to incorporate this into https://github.com/camunda/camunda-platform-docs/pull/2471/commits/0cea47e66891bcc1389575cefd12018a2472c087.

File: docs/components/modeler/bpmn/error-events/error-events.md (hunk 1 above)
Comment (nikku) [reply docs/components/modeler/bpmn/error-events/error-events.md]: I've tried to incorporate this into https://github.com/camunda/camunda-platform-docs/pull/2471/commits/0cea47e66891bcc1389575cefd12018a2472c087.

**Comment by nikku** [comment]
Updated / hopefully simplified this via https://github.com/camunda/camunda-platform-docs/pull/2471/commits/0cea47e66891bcc1389575cefd12018a2472c087 + https://github.com/camunda/camunda-platform-docs/pull/2471/commits/dd66439f2e1c2aa89e7c0e696e99dfcd67399f43.

I removed https://github.com/camunda/camunda-platform-docs/pull/2471/commits/dd66439f2e1c2aa89e7c0e696e99dfcd67399f43 because I think it is redundant; it is clear (indicated through validation + deploy feedback) that the error exists, and we could consider it our baseline.

**Review by korthout — APPROVED** [review]
Love it! 🚀 

LGTM 👍 

💭 the BPMN docs refer to expressions or static values often

🔧 let's create an issue to link all of these statements (no need to clutter this pull request)

>[an `expression` or a static value](/components/concepts/expressions.md#expressions-vs-static-values)

>[a static value](/components/concepts/expressions.md#expressions-vs-static-values)

**Review by christinaausley — APPROVED** [review]
No adjustments on my end for this one 👍  Let me know if you need any assistance with that sitemap error.

File: docs/components/modeler/bpmn/escalation-events/escalation-events.md [hunk 5]
```diff
@@ -73,3 +69,11 @@ An escalation boundary catch event:
 
 <bpmn:escalation id="Escalation_2alpsjo" name="Escalation_2alpsjo" escalationCode="escalationCode" />
 ```
+
+A catch-all escalation boundary catch event:
```
Comment (nikku) [inline docs/components/modeler/bpmn/escalation-events/escalation-events.md:73]:
```suggestion
A escalation boundary catch-all event:
```

File: docs/components/modeler/bpmn/error-events/error-events.md [hunk 6]
```diff
@@ -99,14 +95,13 @@ A boundary error event:
 <bpmn:boundaryEvent id="invalid-credit-card-1" name="Invalid Credit Card" attachedToRef="collect-money">
  <bpmn:errorEventDefinition errorRef="invalid-credit-card-error" />
 </bpmn:boundaryEvent>
-
 ```
 
-A boundary error event without `errorCode`:
+A catch-all error boundary event without:
```
Comment (nikku) [inline docs/components/modeler/bpmn/error-events/error-events.md:100]:
```suggestion
A error boundary catch-all event:
```

**Comment by nikku** [comment]
Fixed one more small oversight - https://github.com/camunda/camunda-platform-docs/pull/2471/commits/9b5e682b7713919b316c94678301cccf8162b23c.

**Comment by nikku** [comment]
Created the linking issue as a potential follow-up: https://github.com/camunda/camunda-platform-docs/issues/2485.

I think it makes sense, and contributes to making the docs more useful, relevant content more accessible.

**Review by remcowesterhoud — APPROVED** [review]

**Pull request merged**